tests/bluetooth/iso/modules/central.c

```diff
@@ -0,0 +1,476 @@
+	k_work_reschedule(&remote_peer[channel_index].dummy_data_send_work, K_MSEC(100));
+}
+
+static void discovery_complete(struct bt_gatt_dm *dm, void *context)
+{
+	int ret;
+	uint8_t channel_index;
+
+	ret = channel_index_get(bt_gatt_dm_conn_get(dm), &channel_index);
+	if (ret) {
+		LOG_ERR("Channel index not found");
+	}
+
+	nus[channel_index] = context;
+
+	ret = bt_nus_handles_assign(dm, nus[channel_index]);
+	if (ret) {
+		LOG_ERR("NUS handle assigne failed, ret = %d", ret);
+	}
+	bt_nus_subscribe_receive(nus[channel_index]);
+	ret = bt_gatt_dm_data_release(dm);
+	if (ret) {
+		LOG_ERR("Release service discovery data failed, ret = %d", ret);
+	}
+
+	k_work_schedule(&remote_peer[channel_index].dummy_data_send_work, K_MSEC(100));
+
+	if (all_peers_connected() == false) {
+		LOG_INF("Not all devices connected");
+		start_scan();
+	} else {
+		LOG_INF("All devices connected");
+		ret = bt_le_scan_stop();
+		if (ret && ret != -EALREADY) {
+			LOG_WRN("Stop scan failed: %d", ret);
+		}
+	}
+}
+
+static void discovery_service_not_found(struct bt_conn *conn, void *context)
+{
+	LOG_ERR("NUS service not found");
+}
+
+static void discovery_error(struct bt_conn *conn, int err, void *context)
+{
+	LOG_ERR("Error while discovering GATT database:, err = %d", err);
+}
+
+struct bt_gatt_dm_cb discovery_cb = {
+	.completed = discovery_complete,
+	.service_not_found = discovery_service_not_found,
+	.error_found = discovery_error,
+};
+
+static int gatt_discover(struct bt_conn *conn)
+{
+	int ret;
+	uint8_t channel_index;
+
+	ret = channel_index_get(conn, &channel_index);
+	if (ret) {
+		LOG_ERR("Channel index not found");
+	}
+
+	ret = bt_gatt_dm_start(conn, BT_UUID_NUS_SERVICE, &discovery_cb,
+			       &nus_client[channel_index]);
+	return ret;
+}
+
+static uint8_t ble_data_received(struct bt_nus_client *nus, const uint8_t *data, uint16_t len)
+{
+	LOG_HEXDUMP_INF(data, len, "NUS received:");
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static int nus_client_init(void)
+{
+	int ret;
+	struct bt_nus_client_init_param init = {.cb = {
+							.received = ble_data_received,
+						}};
+
+	for (int i = 0; i < ARRAY_SIZE(nus_client); i++) {
+		ret = bt_nus_client_init(&nus_client[i], &init);
+		if (ret) {
+			LOG_ERR("NUS Client [%d] initialization failed, ret = %d", i, ret);
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+static int device_found(uint8_t type, const uint8_t *data, uint8_t data_len,
+			const bt_addr_le_t *addr)
+{
+	int ret;
+	struct bt_conn *conn;
+
+	if (all_peers_connected()) {
+		LOG_DBG("All headset connected");
+		return 0;
+	}
+
+	if ((data_len == REMOTE_DEVICE_NAME_PEER_LEN) &&
+	    (strncmp(REMOTE_DEVICE_NAME_PEER, data, REMOTE_DEVICE_NAME_PEER_LEN) == 0)) {
+		LOG_DBG("Device found");
+
+		ret = bt_le_scan_stop();
+		if (ret && ret != -EALREADY) {
+			LOG_WRN("Stop scan failed: %d", ret);
+		}
+
+		ret = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, &conn_param, &conn);
+		if (ret) {
+			LOG_ERR("Could not init connection");
+			start_scan();
```

Comment:
```suggestion
			scan_start();
```
Verb at the end, like what was done with bt_le_scan_stop